flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/StateForTask.java

```diff
@@ -131,11 +108,13 @@ else if (o instanceof StateForTask) {
 
 	@Override
 	public int hashCode() {
-		return state.hashCode() + 31 * operatorId.hashCode() + 43 * subtask;
+		return (int) (this.stateSize ^ this.stateSize >>> 32) +
+			31 * ((int) (this.duration ^ this.duration >>> 32) +
+				31 * state.hashCode());
 	}
 
 	@Override
 	public String toString() {
-		return String.format("StateForTask %s-%d : %s", operatorId, subtask, state);
+		return String.format("StateForTask(%s)", state);
```

Comment:
This was there before as well, but `toString` will always return `StateForTask(SerializedValue)`